applications/StructuralMechanicsApplication/custom_elements/solid_elements/solid_shell_element_sprism_3D6N.cpp

```diff
@@ -4011,26 +4145,159 @@ void SolidShellElementSprism3D6N::CalculateVolumeForce(
+{
+    IntegrationPointsContainerType all_integration_points = AllIntegrationPoints();
+    IntegrationPointsArrayType integration_points = all_integration_points[static_cast<int>(ThisMethod)];
+    // Number of integration points
+    const int integration_points_number = integration_points.size();
+    // Number of nodes in current geometry
+    const int points_number = 6;
+    // Setting up return matrix
+    Matrix shape_function_values( integration_points_number, points_number );
+
+    // Loop over all integration points
+    for ( int pnt = 0; pnt < integration_points_number; pnt++ ) {
+        shape_function_values( pnt, 0 ) = ( 1.0
+                                            - integration_points[pnt].X()
+                                            - integration_points[pnt].Y()
+                                            - integration_points[pnt].Z()
+                                            + ( integration_points[pnt].X() * integration_points[pnt].Z() )
+                                            + ( integration_points[pnt].Y() * integration_points[pnt].Z() ) );
+        shape_function_values( pnt, 1 ) = integration_points[pnt].X()
+                                            - ( integration_points[pnt].X() * integration_points[pnt].Z() );
+        shape_function_values( pnt, 2 ) = integration_points[pnt].Y()
+                                            - ( integration_points[pnt].Y() * integration_points[pnt].Z() );
+        shape_function_values( pnt, 3 ) = integration_points[pnt].Z()
+                                            - ( integration_points[pnt].X() * integration_points[pnt].Z() )
+                                            - ( integration_points[pnt].Y() * integration_points[pnt].Z() );
+        shape_function_values( pnt, 4 ) = ( integration_points[pnt].X() * integration_points[pnt].Z() );
+        shape_function_values( pnt, 5 ) = ( integration_points[pnt].Y() * integration_points[pnt].Z() );
+    }
+
+    return shape_function_values;
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+const SolidShellElementSprism3D6N::ShapeFunctionsValuesContainerType SolidShellElementSprism3D6N::AllShapeFunctionsValues()
+{
+    ShapeFunctionsValuesContainerType shape_functions_values =
+    {
+        {
+            CalculateShapeFunctionsIntegrationPointsValues(GeometryData::IntegrationMethod::GI_GAUSS_1 ),
+            CalculateShapeFunctionsIntegrationPointsValues(GeometryData::IntegrationMethod::GI_GAUSS_2 ),
+            CalculateShapeFunctionsIntegrationPointsValues(GeometryData::IntegrationMethod::GI_GAUSS_3 ),
+            CalculateShapeFunctionsIntegrationPointsValues(GeometryData::IntegrationMethod::GI_GAUSS_4 ),
+            CalculateShapeFunctionsIntegrationPointsValues(GeometryData::IntegrationMethod::GI_GAUSS_5 )
```

Comment:
If you do this we end up with the same mess but in another place... `GI_GAUSS_X` does not refer to Gauss-Legendre integration (as expected) but to a different thing.

I'd just simply hardcode the integration rule you actually need, and remove the possibility to change it as this, as far as I understand, shouldn't be required by the user. Am I wrong?

Reply:
The integration is hardcoded, can be even more hardcoded, but no one will see it. It is not general for all geometries. GeometryData::IntegrationMethod::GI_GAUSS_1 is just an enu,I can simply use a integer.

Reply:
Here you are, now is fully hardcoded

Reply:
Exactly. The deeper the better in this case.

Reply:
Can we please avoid the use of `GeometryData::IntegrationMethod`? The point is that this might be misleading when browsing the code in the future (i.e., I'd expect the standard Gauss rule from the core but we are using a "custom" one in here).

Reply:
I can use an int. It is an enum

![1840c968ae60de0911248c5e334e8ca8de67eaa752c37523812d84409a4e0ec3_1](https://github.com/user-attachments/assets/0262d83e-2c5e-422d-af2f-4f5f82874c74)

Reply:
Yes. So instead of `GI_GAUSS_1` just put `1`. Though equivalent, the message one gets when going through or browsing the code is different.

Reply:
Sure.